.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,361 @@
+name: CaraML CI Pipeline
+
+on:
+  # Automatically run CI on Release and Pre-Release tags and main branch
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+    branches:
+      - main
+    paths-ignore:
+      - "docs/**"
+
+  # Automatically run CI on branches, that have active PR opened
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - "docs/**"
+
+  # To make it possible to trigger e2e CI workflow for any arbitrary git ref
+  workflow_dispatch:
+
+env:
+  GO_VERSION: 1.18
+  NODE_VERSION: 16
+  ARTIFACT_RETENTION_DAYS: 7
+  CONTAINER_REGISTRY: ghcr.io
+  LOCAL_REGISTRY: registry.localhost:5000
+
+jobs:
+  build-ui:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Cache Dependencies
+        uses: actions/cache@v3
+        with:
+          path: ui/node_modules
+          key: |
+            ${{ runner.os }}-modules-${{ hashFiles('ui/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-modules-
+
+      - name: Install
+        working-directory: ui
+        run: yarn install
+
+      - name: Lint code
+        run: make lint-ui
+
+      - name: Build UI
+        env:
+          NODE_OPTIONS: "--max_old_space_size=4096"
+        run: make build-ui
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlp-ui-dist
+          path: ui/build/
+          retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
+
+  build-api:
+    runs-on: ubuntu-latest
+    outputs:
+      api-version: ${{ steps.build-image.outputs.api-version }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Build Docker image
+        id: build-image
+        run: |
+          set -o pipefail
+          make build-api-image | tee output.log
+          echo "api-version=$(sed -n 's%version: \(.*\)%\1%p' output.log)" >> $GITHUB_OUTPUT
+
+      - name: Save Docker image
+        run: |
+          docker image save \
+            --output mlp-api.${{ steps.build-image.outputs.api-version }}.tar \
+            mlp-api:${{ steps.build-image.outputs.api-version }}
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: mlp-api.${{ steps.build-image.outputs.api-version }}.tar
+          path: mlp-api.${{ steps.build-image.outputs.api-version }}.tar
+          retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
+
+  test-api:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/.go
+    services:
+      postgres:
+        image: postgres:12.4
```

Comment:
Why not the latest postgres 12 version: `12.12` ?

Reply:
Why not postgres 14 then? I simply re-used the same version that was defined in the existing workflow. It makes sense to review and bump up the dependencies, tho. Likely, `keto` can be upgraded too.

Reply:
Bumped up postgresql to `14.5`. 
Unfortunately, there are some breaking changes in keto 0.5 API, so simply bumping it up won't work.